index.rst

```diff
@@ -1,61 +1,150 @@
+
+Commissioning
+-------------
+
+In Commissioning, both applying the same processing to different data and applying different processing to the same data will occur frequently.
+Being able to view the definitions of past campaigns and adjust the input datasets or pipeline configurations assists with this.
+
+
+The Scope
+=========
+
+The Campaign Management system sketched in the above use cases serves as a sort of electronic lab notebook, recording parameters and providing links to other systems that do the actual work of executing workflows or rendering plots.
+It is a system to manage campaigns, not one to execute them, although rudimentary execution capabilities (submit the next BPS command in sequence) can be added in later versions.
+Its primary interfaces are human-oriented, not machine-oriented.
+It is usable by staff, including developers, Commissioning staff, and Execution team members.
+It is not initially usable by science users, although with appropriate multi-user security and controls, it could be extended to them.
+
+The generation of inclusion/exclusion lists is acknowledged to require extreme flexibility, with information coming from any available source, including a random number generator.
+As a result, only a format for specifying the exclusion list is defined; the mechanisms and tools for generating such a list are left outside the scope of Campaign Management.
+In particular, the Exposure Log (LSE-490 :cite:`LSE-490` §3.2.4.2 and DMTN-173 :cite:`DMTN-173`) cannot be sufficient to generate inclusion/exclusion lists; it is but one source of information that goes into their generation.
+Inclusion/exclusion lists must be uploaded to the Butler Registry in order to enable joins with other tables and in order to maintain provenance.
+
+
+The Proposed Implementation
+===========================
+
+The Middleware Team will write a tool to upload sets of DataIds in a file in JSON format to a Butler Registry.
+This will require creating a new table in the Registry schema to hold this new "DataId Set" concept; the resulting sets will be persistent and sharable.
+The DataIds must use primary dimension keys in their JSON representation; these can be obtained from the Registry or astro_metadata_translator.
+The Middleware Team may create a batch API to aid in performing this conversion to primary keys, but otherwise it is the user's responsibility.
+
+DataId sets may initially be temporary, but eventually they will have to be persistent as they are used in provenance.
+They will be visible to everyone using the same Registry in initial "friendly user" mode.
+For science users, however sharing permissions, derived collections, and quota issues will need to be worked out.
+
+Note that a DataId list can be turned into a Collection if it is combined with one or more Dataset Types.
```

Comment:
Are we talking only about raw data IDs and exclusion lists here?  There's a lot of language earlier that really pertains only to problems that are intrinsic to raws, and that's what I'd assumed to this point.  If that's the case, we should be explicit about it, and we can just say here that "a raw data ID list is isomorphic to a collection of raws".

If we want talk about applying a raw exclusion list to downstream datasets, I think the appropriate way to do that is to trace the `QuantumGraph` saved in the database (not implemented yet, of course).  That can be saved as a collection, but it's not all you can do with it, and in may cases we'd want to use it to define a modified QG to rebuild the downstream dataset without that raw, instead of a collection of e.g. bad coadd tiles.

Anyhow, if we are instead _also_ considering this system as a way to manage exclusion lists that reject e.g. warps based on the properties of those warps, not the properties of the raws used to build them, as determined from metric values, then I'm much less convinced that an external-to-database list of data IDs is a good model.  Once again, exclusion of the dataset that was identified as bad doesn't necessarily describe what we want to do to fix the problem; if we can trace it back to a raw and mark that as bad for all purposes, we can update the raw exclusions, but in other cases I think we need to update the in-pipeline selection logic when we notice problems (e.g. rejecting visits from coaddition when their PSF models don't meet certain criteria), and either reprocess (time permitting) or declare a known issue in the DR.  Excluding individual downstream datasets explicitly just doesn't scale, because the list isn't persistent across processing runs.